### PR TITLE
Bootstrap fix - move deprecated runtime flag to resource monitors

### PR DESCRIPTION
**Pull request by Pawan-Bishnoi**
Runtime flag global_downstream_max_connections has been deprecated.
Envoy recommends moving to Resource monitors. 
fixes- https://github.com/istio/istio/issues/47355

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
Seems reasonable, will leave to @briansonnenberg or Kuat to approve

File: pkg/bootstrap/testdata/all_golden.json [hunk 1]
```diff
@@ -10,14 +10,23 @@
...
           },
           {
               "name": "admin",
               "admin_layer": {}
           }
       ]
   },
+  "overload_manager": {
+    "resource_monitors": [{
+      "name": "envoy.resource_monitors.downstream_connections",
```
[the remark this answers is not in the document]
Comment (Pawan-Bishnoi) [reply pkg/bootstrap/testdata/all_golden.json:23]: I think we need to add it here: https://github.com/istio/proxy/blob/304387be460e36e265b1fecb21db7d11e6ad3ecc/bazel/extension_config/extensions_build_config.bzl#L189-L194

changing the name didn't fix TestAgent UT

[the remark this answers is not in the document]
Comment (Pawan-Bishnoi) [reply pkg/bootstrap/testdata/all_golden.json:23]: I can raise a PR in istio/proxy if you agree.

[the remark this answers is not in the document]
Comment (Pawan-Bishnoi) [reply pkg/bootstrap/testdata/all_golden.json:23]: plz: https://github.com/istio/proxy/pull/5073

[the remark this answers is not in the document]
Comment (Pawan-Bishnoi) [reply pkg/bootstrap/testdata/all_golden.json]: done, PTAL :)

**Comment by GregHanson** [comment]
not stale

**Comment by hzxuzhonghu** [comment]
https://github.com/envoyproxy/envoy/pull/30620 seems not needed

**Comment by zirain** [comment]
> [envoyproxy/envoy#30620](https://github.com/envoyproxy/envoy/pull/30620) it seems not needed

yes, the new filter still in alpha.

**Comment by Pawan-Bishnoi** [comment]
closing since not needed as of now

**Pull request closed without merging**